pkg/telemetry/telemetryservice.go

```diff
@@ -182,37 +196,23 @@ func (t *telemetryService) getOrCreateWorker(ctx context.Context,
 	)
 
 	t.workers[participantID] = worker
-	t.workersShadow = maps.Values(t.workers)
+
+	worker.next = t.workerList
+	t.workerList = worker
 
 	return worker, false
 }
 
-func (t *telemetryService) cleanupWorkers() {
-	t.lock.RLock()
-	workersShadow := t.workersShadow
-	t.lock.RUnlock()
-
-	toReap := make([]livekit.ParticipantID, 0, len(workersShadow))
-	for _, worker := range workersShadow {
-		closedAt := worker.ClosedAt()
-		if !closedAt.IsZero() && time.Since(closedAt) > workerCleanupWait {
-			worker.Flush()
-
-			toReap = append(toReap, worker.ParticipantID())
-		}
-	}
-
-	if len(toReap) == 0 {
-		return
-	}
+func (t *telemetryService) getAndDeleteWorker(participantID livekit.ParticipantID) (*StatsWorker, bool) {
+	t.workersMu.Lock()
+	worker, ok := t.workers[participantID]
+	delete(t.workers, participantID)
```

Comment:
this will get cleaned out of `workerList` once the close timestamp is set and the cleanup grace period has elapsed